app/controllers/admin/reports_controller.rb

```diff
@@ -46,6 +46,24 @@ def download_user_feedbacks_report
       end
     end
 
+    def download_application_digest_report
+      expires_now
+      respond_to do |format|
+        format.csv do
+          tempfile = Tempfile.new("application_digest_report")
+          headers = ApplicationDigest.first.attributes.keys - %w[id created_at updated_at]
+
+          CSV.open(tempfile, "w", write_headers: true, headers: headers) do |csv|
+            ApplicationDigest.order(create_at: :desc).find_each do |record|
```

Comment:
I think this should be `created_at`. 
```suggestion
            ApplicationDigest.order(created_at: :desc).find_each do |record|
```

I am surprised tests did not breaks as if this was exercised it should have caused an error:
```sh
ApplicationDigest.order(create_at: :desc)
=> An error occurred when inspecting the object: #<ActiveRecord::StatementInvalid....
```

This implies we need more testing around this

Reply:
So it seems this is the default behaviour in Rails, it ignores any custom ordering specified in the query. Therefore, even if there's a typo in the order clause, find_each disregards it and proceeds without raising an error.

So, I can remove the order clause or change to the each method.

I prefer the more performant option, so removing the order clause.

"NOTE: It's not possible to set the order. That is automatically set to ascending on the primary key (“id ASC”) to make the batch ordering work." - https://api.rubyonrails.org/v5.2/classes/ActiveRecord/Batches.html#method-i-find_each

Reply:
That causes us problems as the primary key is a GUID so there is no intrinsic 'order'

I would use Joel's suggestion as that would return the records in an order that makes sense :+1: